app/src/main/java/com/github/se/studybuddies/ui/todo/EditToDo.kt

```diff
@@ -0,0 +1,178 @@
+                        descriptionState,
+                        assigneeState,
+                        onLocationChanged,
+                        locationQuery,
+                        selectedDate,
+                        isOpen)
+                    Button(
+                        onClick = { expanded.value = true },
+                        modifier =
+                            Modifier.padding(0.dp)
+                                .width(300.dp)
+                                .height(45.dp)
+                                .background(Color.Transparent, shape = RoundedCornerShape(10.dp)),
+                        colors = ButtonDefaults.buttonColors(containerColor = Color.Transparent),
+                        border = BorderStroke(1.dp, statusColor(statusState.value))) {
+                          Text(
+                              text = statusState.value.name, color = statusColor(statusState.value))
+                        }
+                    DropdownMenu(
+                        expanded = expanded.value,
+                        onDismissRequest = { expanded.value = false },
+                        modifier =
+                            Modifier.width(300.dp)
+                                .padding(horizontal = 16.dp)
+                                .background(
+                                    color = Color.Transparent, shape = RoundedCornerShape(10.dp))) {
+                          for (status in ToDoStatus.entries) {
+                            DropdownMenuItem(
+                                text = {
+                                  Text(
+                                      status.name,
+                                      textAlign = TextAlign.Center,
+                                      modifier = Modifier.fillMaxWidth())
+                                },
+                                onClick = {
+                                  statusState.value = status
+                                  expanded.value = false
+                                })
+                          }
+                        }
+                    TodoSaveButton(titleState) {
+                      val updatedTodo =
+                          ToDo(
+                              uid = todoUID,
+                              name = titleState.value,
+                              description = descriptionState.value,
+                              assigneeName = assigneeState.value,
+                              dueDate = selectedDate.value,
+                              location = locationState.value,
+                              status = statusState.value)
+                      todoViewModel.updateTodo(todoUID, updatedTodo)
+                      navigationActions.goBack()
+                    }
+                    Button(
+                        onClick = {
+                          todoViewModel.deleteTodo(todoUID)
+                          navigationActions.goBack()

```

Comment:
Try to not used navigationActions.goBack() as it usually end up doing strange go back move (for example delete the taks send you then to the soloStudyScreen instead of the ToDo Screen). For this you should rather use navigationActions.navigateTo(Route.TODOLIST) to be sure that you endup in the Todo Screen and not in another screen,